frontend/packages/console-app/src/components/cloud-shell/cloud-shell-utils.ts

```diff
@@ -96,6 +163,29 @@ export const sendActivityTick = (workspaceName: string, namespace: string): void
 
 export const checkTerminalAvailable = () => coFetch('/api/terminal/available');
 
-export const getCloudShellCR = (name: string, ns: string) => {
-  return k8sGet(WorkspaceModel, name, ns);
+export const getCloudShellCR = (workspaceModel: K8sKind, name: string, ns: string) => {
+  return k8sGet(workspaceModel, name, ns);
+};
+
+// Check to see if v1alpha2 devworkspace crds are available on the cluster.
+// If they are not available or if an error occurs when finding them then report that they aren't available
+export const CheckV1alpha2CRDAvailability = () => {
+  const [loading, setLoading] = useSafetyFirst(true);
+  const [isAvailable, setAvailable] = useSafetyFirst(false);
+
+  React.useEffect(() => {
+    k8sGet(WorkspaceModel)
```

Comment:
Are you trying to fetch a CRD here? This won't work for a normal user.
You should be using feature flags which makes use of API discovery.
You can see how we create the following feature flag in the `console-extensions.json`:
```
  {
    "type": "console.flag/model",
    "properties": {
      "flag": "DEVWORKSPACE",
      "model": {
        "group": "workspace.devfile.io",
        "version": "v1alpha1",
        "kind": "DevWorkspace"
      }
    }
  },
```

Then you can use `useFlag` hook to check the enablement.

Reply:
Ok k8sGet will fetch all resources of that kind.
That's not efficient. So using feature flags is still a better solution.